### ISSUE TITLE
Wayland compatibility issue: screen turns black when taking screenshot

### ISSUE THREAD

**Issue by pieterdd** [martyr-deepin/deepin-screenshot#29]
Something really unusual happens when I start Deepin. It blacks out the screen on which I invoked it. My secondary screen remains in a normal state. The screen clears up when I exit the screenshot tool.

![deepin](https://user-images.githubusercontent.com/1407980/41021124-97c767da-6965-11e8-9a0a-b7f65945b5a4.jpg)

Logs:

> "dtkwidget2" can not find qm files
> "deepin-screenshot" can not find qm files
> 2018-06-06, 08:35:56.768 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.768 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.768 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.768 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.769 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.769 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.772 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.772 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.773 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.773 [Debug  ] [                                                         0] m_keyboardGrabbed: false
> 2018-06-06, 08:35:56.785 [Debug  ] [                                                         0] m_keyboardGrabbed: true
> 2018-06-06, 08:35:56.825 [Debug  ] [                                                         0] ConfigSettings: "effect" "is_blur" QVariant(bool, false)
> 2018-06-06, 08:35:56.831 [Debug  ] [                                                         0] ConfigSettings: "effect" "is_mosaic" QVariant(bool, false)
> 2018-06-06, 08:35:56.831 [Debug  ] [                                                         0] Setting file: "/home/myusername/.config/deepin/deepin-screenshot.conf"
> 2018-06-06, 08:35:56.832 [Debug  ] [                                                         0] ~~~~~~ QSize(276, 28)
> 2018-06-06, 08:35:56.846 [Debug  ] [                                                         0] ConfigSettings: "save" "save_quality" QVariant(int, 100)
> 2018-06-06, 08:35:56.847 [Warning] [                                                         0] QStackedWidget::setCurrentWidget: widget 0x0 not contained in stack
> 

It didn't used to be this way, so I'm wondering what happened? I'm on Antergos (Arch Linux variant) with GNOME 3, using deepin-screenshot 4.0.14-1.

**Comment by pieterdd** [comment]
Turns out it was related to me using Wayland. It works fine under Xorg. I don't know if that's a compatibility issue on their end or yours, so I'll leave this open.

**Comment by rugk** [comment]
Indeed, maybe you'd better rename this issue "Add Wayland compatibility".

Using XWayland tools like Peek (gif screenrecorder) e.g. show it is possible to get compatible.

**Comment by weeman1337** [comment]
Deepin Screenshot is a nice and easy screenshot too.
Would be great to see Wayland support!

**Comment by engelux** [comment]
I have the same issue using fedora 29 with wayland, any update about this? :(

**Comment by qingkong1998** [comment]
是否可以像系统的scrot那样能够获取截图的路径和文件名，而不是给定文件名，系统的scrot截图可以通过提供$n参数来获取截图后的路径，这样便于作进一步处理？

**Comment by tuxedo0801** [comment]
Still no solution for this?! This issue is already about 1year old....

**Comment by BLumia** [comment]
Sorry there are no Wayland supported for the screenshot app yet. Maybe wayland support will be added in future updates.

**Comment by tuxedo0801** [comment]
@BLumia That we already know. Question is: Why? And when will it be fixed?

**Comment by exapsy** [comment]
Still no wayland support :( Can't use deepin on my setup and the issue is 2 years old. Seems like wayland support aint coming soon?

**Comment by tuxedo0801** [comment]
Switched my gnome back to x11. This not only allows the use of deepin-screenshot, but also microsoft teams, which does not support wayland when trying to share the screen.

Sad but true ... :-(
